### ISSUE TITLE
Support for Ignition v0.4.0 and config version "2.0.0"

### ISSUE THREAD

**Issue by dghubble** [coreos/coreos-baremetal#141]
- [x] Allow V2 Ignition YAML
- [x] Allow V2 Ignition JSON
- [x] Migrate Example Profiles ([migration guide](https://github.com/coreos/fuze/blob/master/doc/configuration.md))

Bumped out:
* Respect Ignition Headers (bump to #248, serving machines Ignition 2.0.0 works for higher versions)
* Translate from V1 to V2 if requested (no longer relevant, Fuze YAML does not specify a version)

**Comment by dghubble** [comment]
Blocked by https://github.com/coreos/bugs/issues/1259

**Comment by dghubble** [comment]
The plan is for YAML components of Ignition to be split out into https://github.com/coreos/fuze. Humans will write (or template with bootcfg) fuze configs (a YAML document). Fuze will formalize the transform between a fuze config and different Ignition JSON documents for machines to read.

The changes to `bootcfg` "ignition" templates will be minimal (but breaking) as `bootcfg` has done a similar transform on its own for quite a while. A WIP of these changes can be found in the [fuze-configs](https://github.com/coreos/coreos-baremetal/commits/fuze-configs) branch. This unblocks the migration to the Ignition 2.0.0 format for all example clusters.

If this change is ready, it would be great to include in the coreos-baremetal v0.4.0 release.

cc @crawford
